ibc/types.go

```diff
@@ -205,3 +213,42 @@ type SoftwareUpgradeProposal struct {
 	Height      uint64
 	Info        string // optional
 }
+
+// PoolParams defines parameters for creating a gamm liquidity pool
+type PoolParams struct {
```

Comment:
One source of my confusion is the new types like `ibc.Params`. IBC is used for proposals and software upgrades? I thought these were all done on chain. Can you submit a proposal to another chain via IBC? Do these new types apply to all chains, even non-cosmos?

Reply:
good call, these types belong in `cosmos`